modules/flowable-assertions/flowable-process-assertions/src/main/java/org/flowable/assertions/process/HistoricProcessInstanceAssert.java

```diff
@@ -0,0 +1,150 @@
+import org.flowable.task.api.history.HistoricTaskInstance;
+import org.flowable.variable.api.history.HistoricVariableInstance;
+
+import static org.flowable.assertions.process.FlowableProcessAssertions.assertThat;
+import static org.flowable.assertions.process.Utils.*;
+
+/**
+ * @author martin.grofcik
+ */
+
+public class HistoricProcessInstanceAssert extends AbstractAssert<HistoricProcessInstanceAssert, HistoricProcessInstance> {
+
+    protected final ProcessServicesProvider processServicesProvider;
+
+    protected HistoricProcessInstanceAssert(ProcessEngine processEngine, HistoricProcessInstance historicProcessInstance) {
+        super(historicProcessInstance, HistoricProcessInstanceAssert.class);
+        processServicesProvider = ProcessServicesProvider.of(processEngine);
+    }
+
+    protected HistoricProcessInstanceAssert(HistoricProcessInstance historicProcessInstance) {
+        this(getProcessEngine(), historicProcessInstance);
+    }
+
+    /**
+     * Assert <b>historic</b> activities ordered by activity instance start time.
+     *
+     * @return Assertion of {@link HistoricActivityInstance} list.
+     */
+    public ListAssert<HistoricActivityInstance> activities() {
+        processExistsInHistory();
+
+        return assertThat(processServicesProvider.getHistoryService().createHistoricActivityInstanceQuery().processInstanceId(actual.getId())
+                .orderByHistoricActivityInstanceStartTime().desc().list());
+    }
+
+    /**
+     * Assert <b>historic</b> process instance exists in the history and is finished.
+     *
+     * @return Historic process instance assertion.
+     */
+    public HistoricProcessInstanceAssert isFinished() {
+        processExistsInHistory();
+
+        if (processServicesProvider.getHistoryService().createHistoricProcessInstanceQuery().finished().processInstanceId(actual.getId()).count() != 1) {
+            failWithMessage(getProcessDescription(actual)+" to be finished, but is running in history.");
+        }
+
+        return this;
+    }
+
+    public ListAssert<HistoricVariableInstance> variables() {
+        processExistsInHistory();
+
+        return assertThat(processServicesProvider.getHistoryService().createHistoricVariableInstanceQuery().processInstanceId(actual.getId()).orderByVariableName().asc().list());
+    }
+
+    /**
+     * Assert that process instance has variable in <b>history</b>.
+     *
+     * @param variableName variable to check.
+     * @return Historic process instance assertion
+     */
+
+    public HistoricProcessInstanceAssert hasVariable(String variableName) {
+        processExistsInHistory();
+
+        if (processServicesProvider.getHistoryService().createHistoricProcessInstanceQuery().processInstanceId(actual.getId()).variableExists(variableName).count() != 1) {
+            failWithMessage(getProcessDescription(actual)+" has variable <%s> but variable does not exist in history.", variableName);
```

Comment:
```suggestion
            failWithMessage(getProcessDescription(actual) + " has variable <%s> but variable does not exist in history.", variableName);
```

There are other examples of the same spacing issue.